packages/billing/stripe-webhook-handlers/subscription-deleted.ts

```diff
@@ -71,9 +73,44 @@ export async function handleSubscriptionDeleted(
 
     // This happens when the payment method fails for a couple of times and then Stripe subscription gets expired.
     if (newStatus === 'expired') {
-      // TODO: Put the user back on the free plan (by calling Stripe API). Will be handled in CS-7466
+      await subscribeUserToFreePlan(dbAdapter, event.data.object.customer);
     }
 
     await markStripeEventAsProcessed(dbAdapter, event.id);
   });
 }
+
+async function subscribeUserToFreePlan(
+  dbAdapter: DBAdapter,
+  stripeCustomerId: string,
+) {
+  let stripe = getStripe();
+  let freePlan = await getPlanByMonthlyPrice(dbAdapter, 0);
+  if (!freePlan) {
+    throw new Error('Free plan is not found');
+  }
+  let prices = await stripe.prices.list({
+    product: freePlan.stripePlanId,
+    active: true,
+  });
+  if (!prices.data[0]) {
+    throw new Error('No price found for free plan');
+  }
+
+  try {
+    // After this endpoint is called, Stripe will trigger the `invoice.payment_succeeded` event.
+    // Our webhook handler will process this event to activate the subscription
+    // and grant the user any associated allowances.
+    await stripe.subscriptions.create({
+      customer: stripeCustomerId,
+      items: [
+        {
+          price: prices.data[0].id,
+        },
+      ],
+      payment_behavior: 'error_if_incomplete',
+    });
+  } catch (err) {
+    console.error(`Failed to subscribe user back to free plan, error:`, err);
```

Comment:
Is there a reason this is an error log as opposed to throwing an error? Which are the cases where `stripe.subscriptions.create` can fail, apart from network/server errors?

Reply:
I think we should allow the subscription expiration webhook to succeed, even if we fail to subscribe the user to the free plan.